Demos/Shopping/Shopping/Views/Profile/ProfileView.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  ProfileView.swift
+//  Shopping
+//
+//  Created by Albert Bori on 1/26/23.
+//
+
+import SwiftUI
+import VSM
+
+struct ProfileView: View, ViewStateRendering {
+    typealias Dependencies = ProfileLoaderModel.Dependencies
+    @StateObject var container: StateContainer<ProfileViewState>
+    @State private var username: String = ""
+    
+    init(dependencies: Dependencies) {
+        let loaderModel = ProfileLoaderModel(dependencies: dependencies, error: nil)
+        _container = .init(state: .initialized(loaderModel))
+    }
+    
+    var body: some View {
+        ZStack {
+            switch state {
+            case .initialized(let loaderModel):
+                initializedView(loaderModel: loaderModel)
+            case .loading:
+                ProgressView()
+                    .accessibilityIdentifier("Loading...")
+            case .editing(let editingModel):
+                loadedView(editingModel: editingModel)
+            }
+        }
+        .padding(.all)
+        .navigationTitle("Profile")
+    }
+    
+    @ViewBuilder
+    func initializedView(loaderModel: ProfileLoaderModeling) -> some View {
+        ProgressView()
+            .onAppear {
+                container.observe({ await loaderModel.load() })
+            }
+            .alert(
+                "Oops!",
+                isPresented: .constant(loaderModel.error != nil),
+                presenting: loaderModel.error
+            ) { error in
+                Button("Retry") {
+                    container.observe({ await loaderModel.load() })
+                }
+            } message: { error in
+                Text(error)
+            }
+    }
+    
+    @ViewBuilder
+    func loadedView(editingModel: ProfileEditingModeling) -> some View {
+        VStack(alignment: .leading) {
+            HStack {
+                TextField("User Name", text: $username)
+                    .textFieldStyle(.roundedBorder)
+                    .onAppear {
+                        username = editingModel.username
+                    }
```

Comment:
? - Since this is in `onAppear`, this could potentially be set multiple times throughout a views lifecycle which would cause the user input value to be overwritten.  A couple possible alternatives:
1. I have had success with a custom `onFirstAppearance` view modifier that ensures that the provided closure is only called the first time the view appears.
2. Initialize a Binding with an explicit setter and getter to provide a default to an optional `@State` property that stores the input value.  For example:
```swift
struct Input: View {
    
    let defaultText: String
    @State private var inputText: String?
    
    var body: some View {
        TextField("Input", text: .init(get: {
            inputText ?? defaultText
        }, set: {
            inputText = $0
        }))
    }
}
```